flow/instrumentation/common/resource.go

```diff
@@ -0,0 +1,17 @@
+package common
```

Comment:
avoid creating many small packages, it doesn't really matter until intersubpackage dependencies get involved. go's naming also makes having common names like util, utils, common, etc a pain

Already had to resolve util/utils split

Reply:
the call graph like setup -> tracing/metrics/logging -> common, so had to create a separate one this time.
I'll also change it to otel_common for better visibility